python_modules/dagster/dagster/_daemon/backfill.py

```diff
@@ -140,6 +141,78 @@ def _is_retryable_asset_backfill_error(e: Exception):
     return not isinstance(e, (DagsterError, check.CheckError))
 
 
+def execute_backfill_iteration_with_instigation_logger(
+    backfill: "PartitionBackfill",
+    logger: logging.Logger,
+    workspace_process_context: IWorkspaceProcessContext,
+    instance: "DagsterInstance",
+    debug_crash_flags: Optional[Mapping[str, int]] = None,
+) -> Iterable:
+    with _get_instigation_logger_if_log_storage_enabled(instance, backfill, logger) as _logger:
+        # create a logger that will always include the backfill_id as an `extra`
+        backfill_logger = cast(
+            logging.Logger,
+            logging.LoggerAdapter(_logger, extra={"backfill_id": backfill.backfill_id}),
+        )
```

Comment:
this is more or less identical to before, just moved the logger instantiation inside the method that is potentially called in the threadpool